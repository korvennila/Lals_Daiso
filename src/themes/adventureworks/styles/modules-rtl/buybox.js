import './buybox.scss';
import './storytelling-spotlight.scss';