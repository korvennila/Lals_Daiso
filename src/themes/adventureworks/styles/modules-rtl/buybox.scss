@import "../common/00-settings/colors.scss";
@import "../common/00-settings/icons.scss";
@import "../common/00-settings/responsive-breakpoints.scss";

$buybox-addtocart-margin-left: 8px;
$buybox-addtocart-margin-right: 0;
$media-gallary-padding-left: 18px;
$media-gallary-padding-right: 0;
$msv-buybox-content-width: 30%;
$msv-icon-margin-left: 8px;
$msv-buybox-minmax-label-text-before-padding-left: 5px;
$msv-buybox-msi-exclamation-triangle-before-padding-left: 5px;
$msv-buybox-pickup-option-before-padding-left: 5px;
$msv-buybox-shopsimilarlooks-padding-left: 24px;
$msv-buybox-shopsimilardescriptions-padding-right: 24px;
$rtl-msv-buybox-add-to-wishlist-icon-margin-left: 10px;
$msv-buybox-media-gallery-padding-left: 33px;
$msv-buybox-media-gallery-margin-right: 20px;

[dir="rtl"] {
    .ms-buybox {
        .msc-buybox__bulk-purchase-section {
            .ms-buybox__bulk-purchase-anchor {
                &::before {
                    margin-right: 0;
                    margin-left: 11px;
                }
            }
        }

        &__media-gallery {
            padding-left: $media-gallary-padding-left;
            padding-right: $media-gallary-padding-right;
            margin-top: 48px;

            @media (min-width: $msv-breakpoint-m) and (max-width: $msv-breakpoint-l) {
                padding-left: 0;
            }
        }

        &__content {
            min-width: $msv-buybox-content-width;

            .msc-buybox__pickup-options {
                .ms-buybox__find-in-store-pickup {
                    .ms-buybox__find-in-store-pickup-option::before {
                        padding-left: $msv-buybox-pickup-option-before-padding-left;
                        padding-right: 0;
                    }
                }
            }
        }

        .product-add-to-cart {
            .msc-add-to-cart {
                margin-left: $buybox-addtocart-margin-left;
                margin-right: $buybox-addtocart-margin-right;
            }
        }

        .msc-ss-carousel-vert-strip img {
            display: inline-block;
        }

        .ms-buybox__key_in_price_custom-amount__input {
            text-align: right;
        }

        &__minmax_label_text {
            .msi-InfoCircle {
                margin-left: $msv-icon-margin-left;
            }
        }

        &__quantity {
            .quantity_text_message {
                .ms-buybox__minmax_label_text::before {
                    padding-left: $msv-buybox-minmax-label-text-before-padding-left;
                    padding-right: 0;
                }
            }
        }

        &__add-to-wishlist-container {
            .msc-add-to-wishlist::before {
                margin-right: 0;
                margin-left: $rtl-msv-buybox-add-to-wishlist-icon-margin-left;
            }
        }

        .msc-alert {
            .msc-alert__header {
                .msi-exclamation-triangle::before {
                    padding-left: $msv-buybox-msi-exclamation-triangle-before-padding-left;
                    padding-right: 0;
                }
            }
        }

        .ms-buybox__shopsimilarlooks {
            padding-left: $msv-buybox-shopsimilarlooks-padding-left;
            padding-right: 0;
        }

        .ms-buybox__shopsimilardescriptions {
            padding-right: $msv-buybox-shopsimilardescriptions-padding-right;
            border-right: 0.5px solid $msv-gray-50;
            padding-left: 0;
            border-left: none;
        }

        @media (min-width: ($msv-breakpoint-l)+1) {
            &__media-gallery {
                padding-left: $msv-buybox-media-gallery-padding-left;
                padding-right: 0;
                margin-left: 0;

                .ms-media-gallery {
                    &.vertical {
                        .msc-ss-carousel-vert-slide {
                            .ms-media-gallery__thumbnail-item {
                                margin-left: 0;

                                &:nth-of-type(2n) {
                                    margin-left: 0;
                                    margin-right: $msv-buybox-media-gallery-margin-right;
                                }
                            }
                        }
                    }
                }
            }
        }
    }

    @media (max-width: $msv-breakpoint-l) {
        .ms-media-gallery {
            .msc-carousel {
                &__control {
                    &__prev__icon {
                        @include add-icon($msv-ChevronRight);
                        margin-left: 0;
                    }

                    &__next__icon {
                        @include add-icon($msv-ChevronLeft);
                        margin-right: 0;
                    }
                }
            }
        }
    }

    @media (max-width: $msv-breakpoint-m) {
        .ms-buybox__media-gallery {
            padding-left: 0;
            padding-right: 0;
            margin-left: 0;
        }
    }
}
