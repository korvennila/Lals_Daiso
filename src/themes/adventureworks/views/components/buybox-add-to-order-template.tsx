/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import {
    AddToOrderTemplateComponent,
    IAddToOrderTemplateDialogResources,
    IDuplicateItemsWhenAddingToOrderTemplateDialogResources,
    IItemAddedToOrderTemplateDialogResources,
    IOrderTemplateActionErrorResult,
    IOrderTemplateNameDialogResources
} from '@msdyn365-commerce/components';
import { SimpleProduct } from '@msdyn365-commerce/retail-proxy';
import {
    getConfigureErrors,
    IBuyboxCallbacks,
    IBuyboxCommonResources,
    IBuyboxData,
    IBuyboxState,
    IBuyboxViewProps
} from '@msdyn365-commerce-modules/buybox';
import * as React from 'react';

import { IBuyboxProps as IBuyboxExtentionProps } from '../../definition-extensions/buybox.ext.props.autogenerated';

/**
 * To generate AddToOrderTemplateDialog resources.
 * @param resources - The resources.
 * @returns -- Returns the list of AddToOrderTemplateDialog resources..
 */
function generateAddToOrderTemplateDialogResources(resources: IBuyboxCommonResources): IAddToOrderTemplateDialogResources {
    return {
        addToOrderTemplateHeader: resources.addToOrderTemplateHeader,

        noOrderTemplatesMessage: resources.noOrderTemplatesMessage,
        noOrderTemplatesDescription: resources.noOrderTemplatesDescription,

        createAnOrderTemplateButtonText: resources.createAnOrderTemplateButtonText,
        createNewOrderTemplateButtonText: resources.createNewOrderTemplateButtonText,
        cancelOrderTemplateCreationButtonText: resources.cancelOrderTemplateCreationButtonText,

        selectTemplatesText: resources.selectTemplatesText,
        addToTemplateButtonText: resources.addToTemplateButtonText,
        lineItemsText: resources.lineItemsText
    };
}

/**
 * To generate OrderTemplateNameDialog resources.
 * @param resources - The resources.
 * @returns -- Returns the list of OrderTemplateNameDialog resources..
 */
function generateOrderTemplateNameDialogResources(resources: IBuyboxCommonResources): IOrderTemplateNameDialogResources {
    return {
        orderTemplateHeaderLabel: resources.createOrderTemplateHeader,
        orderTemplateTitleLabel: resources.orderTemplateTitle,
        orderTemplateNameAriaLabel: resources.orderTemplateNameAriaLabel,
        orderTemplateDescription: resources.createOrderTemplateDescription,

        defaultOrderTemplateName: resources.defaultOrderTemplateName,

        orderTemplateButtonLabel: resources.createOrderTemplateButtonText,
        orderTemplateCancelButtonLabel: resources.cancelNewOrderTemplateCreationButtonText
    };
}

/**
 * To generate ItemAddedToOrderTemplateDialog resources.
 * @param resources - The resources.
 * @returns -- Returns the list of ItemAddedToOrderTemplateDialog resources..
 */
function generateItemAddedToOrderTemplateDialogResources(resources: IBuyboxCommonResources): IItemAddedToOrderTemplateDialogResources {
    return {
        viewOrderTemplateButtonText: resources.viewOrderTemplateButtonText,
        continueShoppingButtonText: resources.continueShoppingButtonText,
        itemAddedToOrderTemplateHeaderItemOneText: resources.itemAddedToOrderTemplateHeaderItemOneText,
        itemAddedToOrderTemplateHeaderItemFormatText: resources.itemAddedToOrderTemplateHeaderItemFormatText,
        itemAddedToOrderTemplateHeaderMessageText: resources.itemAddedToOrderTemplateHeaderMessageText,
        freePriceText: resources.priceFree,
        originalPriceText: resources.originalPriceText,
        currentPriceText: resources.currentPriceText
    };
}

/**
 * To generate DuplicateItemsWhenAddingToOrderTemplateDialog resources.
 * @param resources - The resources.
 * @returns -- Returns the list of DuplicateItemsWhenAddingToOrderTemplateDialog resources..
 */
function generateDuplicateItemsWhenAddingToOrderTemplateDialogResources(
    resources: IBuyboxCommonResources
): IDuplicateItemsWhenAddingToOrderTemplateDialogResources {
    return {
        duplicatedProductsHeader: resources.duplicatedProductsHeader,
        duplicatedProductsDescription: resources.duplicatedProductsDescription,
        updateQuantityButtonText: resources.updateQuantityButtonText,
        cancelDuplicateItemsButtonText: resources.cancelDuplicateItemsButtonText
    };
}

/**
 * Add to wishlist failed function.
 * @param callbacks -Buybox callbacks.
 * @param resources -Buybox resources.
 * @param product -Simple product.
 * @returns Update error state.
 */
const onAddToOrderTemplateFailed = (callbacks: IBuyboxCallbacks, resources: IBuyboxCommonResources, product: SimpleProduct | undefined) => (
    result: IOrderTemplateActionErrorResult
) => {
    callbacks.updateErrorState({
        errorHost: 'ORDER_TEMPLATE',
        configureErrors:
            result.status === 'MISSING_DIMENSION' ? getConfigureErrors(result.missingDimensions, resources, product?.IsGiftCard) : {}
    });
};

/**
 * RenderAddToOrderTemplateButton.
 * @param props - Buybox view props.
 * @param state - Buybox state.
 * @param callbacks - Buybox callbacks.
 * @param product - Product data.
 * @returns -- Returns JSX element.
 */
export const renderAddToOrderTemplateButton = (
    props: IBuyboxViewProps & IBuyboxExtentionProps<IBuyboxData>,
    state: IBuyboxState,
    callbacks: IBuyboxCallbacks,
    product: SimpleProduct
): React.ReactNode => {
    const {
        // eslint-disable-next-line @typescript-eslint/naming-convention -- Dependency from module file.
        id,
        typeName,
        context,
        data: {
            orderTemplates: { result: orderTemplates },
            customerInformation: { result: customerInformation }
        },
        resources,
        config
    } = props;

    const { quantity } = state;

    const isShowButtonText = true;
    const isMultipleCatalogsFeatureEnabled = props.isChannelMultipleCatalogsFeatureEnabled || false;
    return (
        orderTemplates && (
            <AddToOrderTemplateComponent
                className='msc-add-to-cart-extra-actions'
                addToOrderTemplateButtonText={resources.addToOrderTemplateButtonText}
                addToOrderTemplateButtonTooltip={resources.addToOrderTemplateButtonTooltip}
                addToOrderTemplateDialogResources={generateAddToOrderTemplateDialogResources(resources)}
                createOrderTemplateDialogResources={generateOrderTemplateNameDialogResources(resources)}
                itemAddedToOrderTemplateDialogResources={generateItemAddedToOrderTemplateDialogResources(resources)}
                duplicateItemsWhenAddingToOrderTemplateDialogResources={generateDuplicateItemsWhenAddingToOrderTemplateDialogResources(
                    resources
                )}
                data={{ product, quantity, orderTemplates, customerInformation }}
                context={context}
                id={id}
                typeName={typeName}
                onError={onAddToOrderTemplateFailed(callbacks, resources, product)}
                shouldShowButtonFailedTooltip={state.isCustomPriceSelected}
                showButtonTooltip={false}
                showButtonText={isShowButtonText}
                imageSettings={config && config.imageSettings !== undefined ? config.imageSettings : undefined}
                isMultipleCatalogFeatureEnabledInHq={isMultipleCatalogsFeatureEnabled}
                {...(isMultipleCatalogsFeatureEnabled && { catalogId: props.currentCatalogId })}
            />
        )
    );
};
