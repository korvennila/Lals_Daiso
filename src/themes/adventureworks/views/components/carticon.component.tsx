/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import { IComponent, IComponentProps, msdyn365Commerce } from '@msdyn365-commerce/core';
import { ICartState } from '@msdyn365-commerce/global-state';
import { format, getPayloadObject, getTelemetryAttributes, ITelemetryContent } from '@msdyn365-commerce-modules/utilities';
import { observer } from 'mobx-react';
import * as React from 'react';

/**
 * ICartIconData: Interface for Cart Icon Data.
 */
export interface ICartIconData {
    cart?: ICartState;
}

/**
 * ICartIconComponentProps: Interface for Cart Icon component props.
 */
export interface ICartIconComponentProps extends IComponentProps<ICartIconData> {
    className?: string;
    cartLabel: string;
    cartQtyLabel: string;
    telemetryContent?: ITelemetryContent;
    salesAgreementPricePrompt?: string;
}

/**
 * ICartIconComponent: Interface for Cart Icon component.
 */
export interface ICartIconComponent extends IComponent<ICartIconComponentProps> {}

const CartIconComponentActions = {};

/**
 *
 * CartIcon component.
 * @extends {React.FC<ICartIconComponentProps>}
 */
const CartIcon: React.FC<ICartIconComponentProps> = observer((props: ICartIconComponentProps) => {
    const {
        cartLabel,
        cartQtyLabel,
        data: { cart }
    } = props;
    const defaultCartItemCount: number = 0;

    const cartItem = cart ? cart.totalItemsInCart : defaultCartItemCount;
    let qtyLabel: string = '';
    const cartMaxQuantity: number = 99;
    const maxQuantityLabel: string = '99+';
    if (cartItem > cartMaxQuantity) {
        qtyLabel = maxQuantityLabel;
    } else {
        qtyLabel = format(cartQtyLabel, cartItem);
    }
    const label = format(cartLabel, cartItem);

    // Construct telemetry attribute to render
    const payLoad = getPayloadObject('click', props.telemetryContent!, 'cart-icon', '');
    const attributes = getTelemetryAttributes(props.telemetryContent!, payLoad);
    const style: React.CSSProperties = {
        visibility: cart ? 'visible' : 'hidden'
    };
    return (
        <div className='msc-cart-icon' role='button' aria-label={label} title={label} {...attributes}>
            <div className='msc-cart-icon__count' aria-hidden style={style}>
                {qtyLabel}
            </div>
        </div>
    );
});

export const CartIconComponent: React.FunctionComponent<ICartIconComponentProps> = msdyn365Commerce.createComponentOverride<
    // @ts-expect-error -- Compatible issue with the component override.
    ICartIconComponent
>('CartIcon', { component: CartIcon, ...CartIconComponentActions });

export default CartIconComponent;
