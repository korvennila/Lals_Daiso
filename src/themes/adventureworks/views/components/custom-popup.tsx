import React from 'react';

interface PopupProps {
    message: string;
    isVisible: boolean;
    onClose: () => void;
}

const CustomPopup: React.FC<PopupProps> = ({ message, isVisible, onClose }) => {
    if (!isVisible) return null;

    return (
        <div className='custom-popup-container'>
            <div className='custom-popup-content'>
                <p>{message}</p>
                <button onClick={onClose}>Close</button>
            </div>
            <div className='custom-popup-overlay' onClick={onClose}></div>
        </div>
    );
};

export default CustomPopup;
