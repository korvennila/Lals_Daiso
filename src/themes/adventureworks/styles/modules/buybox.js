import './buybox.scss';
import './storytelling-spotlight.scss';
import './swatch.scss';