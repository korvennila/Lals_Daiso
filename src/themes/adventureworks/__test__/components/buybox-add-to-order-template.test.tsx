/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import { IOrderTemplateActionErrorResult } from '@msdyn365-commerce/components';
import { ICoreContext } from '@msdyn365-commerce/core';
import { AsyncResult, Customer, ProductListLine, SimpleProduct } from '@msdyn365-commerce/retail-proxy';
import { IBuyboxData, IBuyboxState, IBuyboxViewProps } from '@msdyn365-commerce-modules/buybox';
import * as RetailActions from '@msdyn365-commerce-modules/retail-actions';

import { IBuyboxProps as IBuyboxExtentionProps, IBuyboxResources } from '../../definition-extensions/buybox.ext.props.autogenerated';
import { renderAddToOrderTemplateButton } from '../../views/components/buybox-add-to-order-template';

describe('Add To Order Template Button unit tests - View', () => {
    const productListLine: ProductListLine = {
        ProductListId: 'product-list-id',
        LineId: 'line-id',
        ProductId: 42,
        Quantity: 12,
        UnitOfMeasure: 'ea'
    };

    const emptyOrderTemplate: RetailActions.OrderTemplate = {
        productList: {
            Id: 'id'
        },
        orderTemplateLines: [],
        totalLines: 0
    };

    const orderTemplatesT: RetailActions.OrderTemplate[] = [
        {
            productList: {
                Id: 'id',
                ProductListLines: [productListLine]
            },
            orderTemplateLines: [{ productListLine }],
            totalLines: 0
        },
        emptyOrderTemplate
    ];

    const b2bCustomer: Customer = {
        AccountNumber: 'customer-id',
        IsB2b: true
    };

    // @ts-expect-error partial mock
    const mockData: IBuyboxData = {
        orderTemplates: {
            status: 'FAILED',
            result: orderTemplatesT
        } as AsyncResult<RetailActions.OrderTemplate[]>,
        customerInformation: {
            status: 'SUCCESS',
            result: b2bCustomer
        } as AsyncResult<Customer>
    };

    // @ts-expect-error partial mock
    const mockData1: IBuyboxData = {
        orderTemplates: {
            status: 'FAILED',
            result: undefined
        } as AsyncResult<RetailActions.OrderTemplate[]>,
        customerInformation: {
            status: 'SUCCESS',
            result: undefined
        } as AsyncResult<Customer>
    };

    // @ts-expect-error partial mock
    const mockContext: ICoreContext = {};

    const mockResources: IBuyboxResources = {
        findInStoreLinkText: 'ms-buybox',
        findInStoreHeaderText: 'ms-buybox',
        findInStoreDescriptionText: 'ms-buybox',
        priceFree: 'ms-buybox',
        originalPriceText: 'ms-buybox',
        currentPriceText: 'ms-buybox',
        addToCartText: 'ms-buybox',
        outOfStockText: 'ms-buybox',
        averageRatingAriaLabel: 'ms-buybox',
        selectDimensionFormatString: 'ms-buybox',
        productDimensionTypeColor: 'ms-buybox',
        productDimensionTypeConfiguration: 'ms-buybox',
        productDimensionTypeSize: 'ms-buybox',
        productDimensionTypeStyle: 'ms-buybox',
        productDimensionTypeAmount: 'ms-buybox',
        createOrderTemplateHeader: 'ms-buybox',
        orderTemplateTitle: 'ms-buybox',
        orderTemplateNameAriaLabel: 'ms-buybox',
        createOrderTemplateDescription: 'ms-buybox',
        createNewOrderTemplateButtonText: 'ms-buybox',
        createOrderTemplateButtonText: 'ms-buybox',
        cancelNewOrderTemplateCreationButtonText: 'ms-buybox',
        defaultOrderTemplateName: 'ms-buybox',
        addToOrderTemplateHeader: 'ms-buybox',
        addToOrderTemplateButtonTooltip: 'ms-buybox',
        noOrderTemplatesMessage: 'ms-buybox',
        noOrderTemplatesDescription: 'ms-buybox',
        createAnOrderTemplateButtonText: 'ms-buybox',
        cancelOrderTemplateCreationButtonText: 'ms-buybox',
        selectTemplatesText: 'ms-buybox',
        addToTemplateButtonText: 'ms-buybox',
        lineItemsText: 'ms-buybox',
        addToOrderTemplateButtonText: 'ms-buybox',
        addToOrderTemplateMessage: 'ms-buybox',
        addItemToOrderTemplateError: 'ms-buybox',
        viewOrderTemplateButtonText: 'ms-buybox',
        continueShoppingButtonText: 'ms-buybox',
        itemAddedToOrderTemplateHeaderItemOneText: 'ms-buybox',
        itemAddedToOrderTemplateHeaderItemFormatText: 'ms-buybox',
        itemAddedToOrderTemplateHeaderMessageText: 'ms-buybox',
        duplicatedProductsHeader: 'ms-buybox',
        duplicatedProductsDescription: 'ms-buybox',
        updateQuantityButtonText: 'ms-buybox',
        cancelDuplicateItemsButtonText: 'ms-buybox',
        addToWishlistButtonText: 'ms-buybox',
        removeFromWishlistButtonText: 'ms-buybox',
        nameOfWishlist: 'ms-buybox',
        inputQuantityAriaLabel: 'ms-buybox',
        addToWishlistMessage: 'ms-buybox',
        removedFromWishlistMessage: 'ms-buybox',
        addItemToWishlistError: 'ms-buybox',
        removeItemFromWishlistError: 'ms-buybox',
        productQuantityHeading: 'ms-buybox',
        errorMessageOutOfStock: 'ms-buybox',
        errorMessageOutOfRangeOneLeft: 'ms-buybox',
        errorMessageOutOfRangeFormat: 'ms-buybox',
        productDimensionTypeColorErrorMessage: 'ms-buybox',
        productDimensionTypeConfigurationErrorMessage: 'ms-buybox',
        productDimensionTypeSizeErrorMessage: 'ms-buybox',
        productDimensionTypeStyleErrorMessage: 'ms-buybox',
        productDimensionTypeAmountErrorMessage: 'ms-buybox',
        buyboxErrorMessageHeader: 'ms-buybox',
        addedToCartFailureMessage: 'ms-buybox',
        maxQuantityLimitText: 'ms-buybox',
        invalidSmallCustomAmountText: 'ms-buybox',
        invalidLargeCustomAmountText: 'ms-buybox',
        shopSimilarLooksText: 'ms-buybox',
        shopSimilarDescriptionText: 'ms-buybox',
        buyBoxGoToCartText: 'ms-buybox',
        buyBoxContinueShoppingText: 'ms-buybox',
        buyBoxSingleItemText: 'ms-buybox',
        buyBoxMultiLineItemFormatText: 'ms-buybox',
        buyBoxHeaderMessageText: 'ms-buybox',
        maxQuantityText: 'ms-buybox',
        minQuantityText: 'ms-buybox',
        addedQuantityText: 'ms-buybox',
        buyboxKeyInPriceLabelHeading: 'ms-buybox',
        descriptionTextToShowAllPickupOptions: 'ms-buybox',
        closeNotificationLabel: 'ms-buybox',
        findInStoreNotAvailableTextButton: 'ms-buybox',
        findInStoreNotAvailableDescription: 'ms-buybox',
        ariaLabelForSelectedSwatchValue: 'ms-buybox',
        informationIconText: 'ms-buybox',
        priceRangeSeparator: 'ms-buybox',
        availabilityAtPreferredStoreText: 'Availability at preferred store',
        inStockText: 'In stock',
        bulkPurchaseLinkText: 'Bulk Purchase Link Text',
        swatchItemAriaLabel: 'Product',
        priceText: 'Price',
        shopText: 'Shop',
        descriptionText: 'Description',
        findText: 'Pickup and Delivery Methods',
        skuText: 'SKU:',
        decrementButtonAriaLabel: 'Decrease the quantity by 1',
        incrementButtonAriaLabel: 'Increase quantity by 1',
        salesAgreementPricePrompt: 'Contract price and terms may apply.',
        salesAgreementExpirationDatePrompt: 'Available period: {0} - {1}',
        salesAgreementCommittedQuantityPrompt: 'Quantity commitment: {0}',
        salesAgreementRemainingQuantityPrompt: 'Remaining quantity to qualify: {0}'
    };

    // @ts-expect-error partial mock
    const mockState: IBuyboxState = {
        quantity: 1
    };

    // @ts-expect-error partial mock
    const mockEmptyState: IBuyboxState = {};

    // @ts-expect-error partial mock
    const mockProps: IBuyboxViewProps & IBuyboxExtentionProps<IBuyboxData> = {
        id: '123',
        typeName: 'type name',
        context: mockContext,
        data: mockData,
        resources: mockResources,
        config: {
            imageSettings: undefined
        }
    };

    const mockCallbacks = {
        updateQuantity: jest.fn(),
        updateKeyInPrice: jest.fn(),
        updateErrorState: jest.fn(),
        initializeDimension: jest.fn(),
        dimensionSelectedAsync: jest.fn(),
        updateSelectedProduct: jest.fn(),
        getDropdownName: jest.fn(),
        changeModalOpen: jest.fn(),
        changeUpdatingDimension: jest.fn()
    };

    const mockProduct: SimpleProduct = {
        Name: 'Crew neck server',
        RecordId: 123,
        MasterProductId: 123,
        BasePrice: 25,
        Price: 25,
        AdjustedPrice: 25,
        ProductTypeValue: 1,
        Dimensions: [
            {
                DimensionTypeValue: 3,
                DimensionValue: {
                    RecordId: 1,
                    Value: 'S'
                }
            },
            {
                DimensionTypeValue: 4,
                DimensionValue: {
                    RecordId: 3,
                    Value: 'Large'
                }
            }
        ]
    };

    it('renders correctly', () => {
        const component = renderAddToOrderTemplateButton(mockProps, mockState, mockCallbacks, mockProduct) as React.Component<
            IBuyboxViewProps & IBuyboxExtentionProps<IBuyboxData>
        >;

        const errorResult: IOrderTemplateActionErrorResult = {
            status: 'FAILED'
        };

        // @ts-expect-error
        component.props.onError(errorResult);

        errorResult.status = 'MISSING_DIMENSION';

        // @ts-expect-error
        component.props.onError(errorResult);

        expect(component).toMatchSnapshot();
    });

    it('renders correctly with mock data 1', () => {
        mockProps.data = mockData1;
        const component = renderAddToOrderTemplateButton(mockProps, mockEmptyState, mockCallbacks, mockProduct);
        expect(component).toMatchSnapshot();
    });

    it('renders correctly mock empty', () => {
        const component = renderAddToOrderTemplateButton(mockProps, mockEmptyState, mockCallbacks, mockProduct);
        expect(component).toMatchSnapshot();
    });
});
