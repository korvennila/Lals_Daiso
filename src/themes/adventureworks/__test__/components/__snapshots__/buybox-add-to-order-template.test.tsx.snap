// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add To Order Template Button unit tests - View renders correctly 1`] = `
<Unknown
  addToOrderTemplateButtonText="ms-buybox"
  addToOrderTemplateButtonTooltip="ms-buybox"
  addToOrderTemplateDialogResources={
    Object {
      "addToOrderTemplateHeader": "ms-buybox",
      "addToTemplateButtonText": "ms-buybox",
      "cancelOrderTemplateCreationButtonText": "ms-buybox",
      "createAnOrderTemplateButtonText": "ms-buybox",
      "createNewOrderTemplateButtonText": "ms-buybox",
      "lineItemsText": "ms-buybox",
      "noOrderTemplatesDescription": "ms-buybox",
      "noOrderTemplatesMessage": "ms-buybox",
      "selectTemplatesText": "ms-buybox",
    }
  }
  className="msc-add-to-cart-extra-actions"
  context={Object {}}
  createOrderTemplateDialogResources={
    Object {
      "defaultOrderTemplateName": "ms-buybox",
      "orderTemplateButtonLabel": "ms-buybox",
      "orderTemplateCancelButtonLabel": "ms-buybox",
      "orderTemplateDescription": "ms-buybox",
      "orderTemplateHeaderLabel": "ms-buybox",
      "orderTemplateNameAriaLabel": "ms-buybox",
      "orderTemplateTitleLabel": "ms-buybox",
    }
  }
  data={
    Object {
      "customerInformation": Object {
        "AccountNumber": "customer-id",
        "IsB2b": true,
      },
      "orderTemplates": Array [
        Object {
          "orderTemplateLines": Array [
            Object {
              "productListLine": Object {
                "LineId": "line-id",
                "ProductId": 42,
                "ProductListId": "product-list-id",
                "Quantity": 12,
                "UnitOfMeasure": "ea",
              },
            },
          ],
          "productList": Object {
            "Id": "id",
            "ProductListLines": Array [
              Object {
                "LineId": "line-id",
                "ProductId": 42,
                "ProductListId": "product-list-id",
                "Quantity": 12,
                "UnitOfMeasure": "ea",
              },
            ],
          },
          "totalLines": 0,
        },
        Object {
          "orderTemplateLines": Array [],
          "productList": Object {
            "Id": "id",
          },
          "totalLines": 0,
        },
      ],
      "product": Object {
        "AdjustedPrice": 25,
        "BasePrice": 25,
        "Dimensions": Array [
          Object {
            "DimensionTypeValue": 3,
            "DimensionValue": Object {
              "RecordId": 1,
              "Value": "S",
            },
          },
          Object {
            "DimensionTypeValue": 4,
            "DimensionValue": Object {
              "RecordId": 3,
              "Value": "Large",
            },
          },
        ],
        "MasterProductId": 123,
        "Name": "Crew neck server",
        "Price": 25,
        "ProductTypeValue": 1,
        "RecordId": 123,
      },
      "quantity": 1,
    }
  }
  duplicateItemsWhenAddingToOrderTemplateDialogResources={
    Object {
      "cancelDuplicateItemsButtonText": "ms-buybox",
      "duplicatedProductsDescription": "ms-buybox",
      "duplicatedProductsHeader": "ms-buybox",
      "updateQuantityButtonText": "ms-buybox",
    }
  }
  id="123"
  isMultipleCatalogFeatureEnabledInHq={false}
  itemAddedToOrderTemplateDialogResources={
    Object {
      "continueShoppingButtonText": "ms-buybox",
      "currentPriceText": "ms-buybox",
      "freePriceText": "ms-buybox",
      "itemAddedToOrderTemplateHeaderItemFormatText": "ms-buybox",
      "itemAddedToOrderTemplateHeaderItemOneText": "ms-buybox",
      "itemAddedToOrderTemplateHeaderMessageText": "ms-buybox",
      "originalPriceText": "ms-buybox",
      "viewOrderTemplateButtonText": "ms-buybox",
    }
  }
  onError={[Function]}
  showButtonText={true}
  showButtonTooltip={false}
  typeName="type name"
/>
`;

exports[`Add To Order Template Button unit tests - View renders correctly mock empty 1`] = `undefined`;

exports[`Add To Order Template Button unit tests - View renders correctly with mock data 1 1`] = `undefined`;
